### DISCUSSION TITLE
This thing is awesome!

### DISCUSSION THREAD

**Discussion by BenjiDQ** [InterwebAlchemy/obsidian-ai-research-assistant#19] [General]
I have a few ideas to make this better, but no coding skills to contribute to the project so will keep them to myself. Really love using this tool, though I don't use it for research, have another use case for it.

**Comment by ericrallen** [comment]
Thanks for the kind words!

Feel encouraged to share some of your ideas here. If they fit the theme of the project, we can convert them into Issues and someone might pick them up to turn them in to a reality.

I have a lot more ideas for it, too, but only get to spend the time to work on a few of them.

**Reply by BenjiDQ** [reply]
These are the two main features that I think would elevate this to the next level:

1. Presets for different preambles. I have a variety of different preambles that I have to copy-paste per conversation. 
2. The ability to customize the formatting of the saved notes. I am curious if there is a file I could edit to achieve this since I only need one format. After the conversation, I often edit the saved notes to include my original prompt, YAML, and the outputted message. It would be a godsend if those sections could be added automatically when creating the note.

**Reply by ericrallen** [reply]
Thanks for the feedback!

1. I have definitely had "save conversation settings as presets" on my todo list for a long time
2. This is currently handled by [`src/templates/summaryTemplate.ts`](https://github.com/InterwebAlchemy/obsidian-ai-research-assistant/blob/main/src/templates/summaryTemplate.ts), but I would like to move to some sort of user-controlled templates directory that allows you to manipulate the output and choose which template should be used to render it. 

Number 2 is further down the road because I feel like I need to solve incrementally updating the saved conversation instead of overwriting it and also loading a conversation from the stored conversation before I can focus on the display of it.

If you don't mind sharing a simple example of the original AI Research Assitant saved note and then your updated version with the details you want, I can see if it would be much of an effort to just add the functionality you're handling manually right now until I can get around to the user-generated templates.
